src/Neo/SmartContract/Native/NativeContract.cs

```diff
@@ -288,13 +290,19 @@ internal bool IsInitializeBlock(ProtocolSettings settings, uint index, out Hardf
 
                 if (activeIn == index)
                 {
-                    hardfork = hf;
-                    return true;
+                    hfs.Add(hf);
                 }
             }
```

Comment:
Remove from line `283-295` and replace with:
```csharp
hfs = [.. usedHardforks.Where((v, i) => activeIn == i)];
```

Reply:
looks slower

Reply:
what? Its optimized. Compiler writes less `opcodes`.

Reply:
Linq always is slower

Reply:
Not as slow of the lines above it, that it would be replacing. Your right in the way that doing `usedHardforks.Any()` vs `usedHardforks.Length == 0` would be fast for sure.